presto-oracle/src/test/java/io/prestosql/plugin/oracle/TestOracleDistributedQueries.java

```diff
@@ -0,0 +1,227 @@
+    protected boolean supportsViews()
+    {
+        return false;
+    }
+
+    @Override
+    protected boolean supportsArrays()
+    {
+        return false;
+    }
+
+    @Override
+    public void testCommentTable()
+    {
+        // table comment not supported
+    }
+
+    @Override
+    public void testDelete()
+    {
+        // delete is not supported
+    }
+
+    @Test
+    @Override
+    public void testLargeIn()
+    {
+        Integer numberOfElements = 1000;
+        String longValues = range(0, numberOfElements)
+                .mapToObj(Integer::toString)
+                .collect(joining(", "));
+        assertQuery("SELECT orderkey FROM orders WHERE orderkey IN (" + longValues + ")");
+        assertQuery("SELECT orderkey FROM orders WHERE orderkey NOT IN (" + longValues + ")");
+
+        assertQuery("SELECT orderkey FROM orders WHERE orderkey IN (mod(1000, orderkey), " + longValues + ")");
+        assertQuery("SELECT orderkey FROM orders WHERE orderkey NOT IN (mod(1000, orderkey), " + longValues + ")");
+
+        String arrayValues = range(0, numberOfElements)
+                .mapToObj(i -> format("ARRAY[%s, %s, %s]", i, i + 1, i + 2))
+                .collect(joining(", "));
+        assertQuery("SELECT ARRAY[0, 0, 0] in (ARRAY[0, 0, 0], " + arrayValues + ")", "values true");
+        assertQuery("SELECT ARRAY[0, 0, 0] in (" + arrayValues + ")", "values false");
+    }
+
+    @Test
+    @Override
+    public void testCreateTable()
+    {
+        assertUpdate("CREATE TABLE test_create (a bigint, b double, c varchar)");
+        assertTrue(getQueryRunner().tableExists(getSession(), "test_create"));
+        assertTableColumnNames("test_create", "a", "b", "c");
+
+        assertUpdate("DROP TABLE test_create");
+        assertFalse(getQueryRunner().tableExists(getSession(), "test_create"));
+
+        assertQueryFails("CREATE TABLE test_create (a bad_type)", ".* Unknown type 'bad_type' for column 'a'");
+        assertFalse(getQueryRunner().tableExists(getSession(), "test_create"));
+
+        // reduced the table name to 30 chars, from test_create_table_if_not_exists to test_create_table_if_not_exist
+        assertUpdate("CREATE TABLE test_create_table_if_not_exist (a bigint, b varchar, c double)");
+        assertTrue(getQueryRunner().tableExists(getSession(), "test_create_table_if_not_exist"));
+        assertTableColumnNames("test_create_table_if_not_exist", "a", "b", "c");
+
+        assertUpdate("CREATE TABLE IF NOT EXISTS test_create_table_if_not_exist (d bigint, e varchar)");
+        assertTrue(getQueryRunner().tableExists(getSession(), "test_create_table_if_not_exist"));
+        assertTableColumnNames("test_create_table_if_not_exist", "a", "b", "c");
+
+        assertUpdate("DROP TABLE test_create_table_if_not_exist");
+        assertFalse(getQueryRunner().tableExists(getSession(), "test_create_table_if_not_exist"));
+
+        // I comment this because LIKE is a reserved word on oracle and is failing
```

Comment:
CREATE TABLE .. LIKE  is evaluated on Presto side, so it should not matter whether this is a keyword or not in oracle. What am i missing?

Reply:
you are correct..
it's the boolean type that is not mapped..

but in oracle this dont exist 
one way to have "boolean" on oracle is this: "value NUMBER(1) CHECK (value IN (0,1))"

but doing this will create more problems on toPrestoType.. 
as we are mapping boolean, tinyint, smallint, integer and bigint as numeric